src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialsStore.java

```diff
@@ -0,0 +1,135 @@
+        return Jenkins.getInstance().getACL().hasPermission(authentication, permission);
+    }
+
+    @NonNull
+    @Override
+    public List<Credentials> getCredentials(@NonNull Domain domain) {
+        // TODO: Filter by domain - how do I do this?
+        return  provider.getCredentials(Credentials.class, Jenkins.getInstance(), ACL.SYSTEM);
+    }
+
+    @Override
+    public boolean addCredentials(@NonNull Domain domain, @NonNull Credentials credentials) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public boolean removeCredentials(@NonNull Domain domain, @NonNull Credentials credentials) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public boolean updateCredentials(@NonNull Domain domain, @NonNull Credentials current,
+                                     @NonNull Credentials replacement) {
+        throw new NotImplementedException();
+    }
+
+    @Nullable
+    @Override
+    public CredentialsStoreAction getStoreAction() {
+        return action;
+    }
+
+    /**
+     * Expose the store.
+     */
+    @ExportedBean
+    public static class KubernetesCredentialsStoreAction extends CredentialsStoreAction {
+
+        private final KubernetesCredentialsStore store;
+
+        private KubernetesCredentialsStoreAction(KubernetesCredentialsStore store) {
+            this.store = store;
+            addIcons();
+        }
+
+        private void addIcons() {
+            IconSet.icons.addIcon(new Icon("icon-credentials-kubernetes-store icon-sm",
+                "kubernetes-credentials-provider/images/16x16/kubernetes-store.png",
+                Icon.ICON_SMALL_STYLE, IconType.PLUGIN));
+            IconSet.icons.addIcon(new Icon("icon-credentials-kubernetes-store icon-md",
+                "kubernetes-credentials-provider/images/24x24/kubernetes-store.png",
+                Icon.ICON_MEDIUM_STYLE, IconType.PLUGIN));
+            IconSet.icons.addIcon(new Icon("icon-credentials-kubernetes-store icon-lg",
+                "kubernetes-credentials-provider/images/32x32/kubernetes-store.png",
+                Icon.ICON_LARGE_STYLE, IconType.PLUGIN));
+            IconSet.icons.addIcon(new Icon("icon-credentials-kubernetes-store icon-xlg",
+                "kubernetes-credentials-provider/images/48x48/kubernetes-store.png",
+                Icon.ICON_XLARGE_STYLE, IconType.PLUGIN));
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @NonNull
+        public CredentialsStore getStore() {
+            return store;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String getIconFileName() {
+            return isVisible()
+               ? "/plugin/kubernetes-credentials-provider/images/32x32/kubernetes-store.png"
+               : null;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String getIconClassName() {
+            return isVisible()
```

Comment:
what is isVisible?

Reply:
TBH, not sure. I followed https://github.com/jenkinsci/credentials-plugin/blob/e03e9b30d62d4683d587fa6a775904e089296e9a/src/main/java/com/cloudbees/plugins/credentials/SystemCredentialsProvider.java#L628 as an example